src/functions/Coverage.ps1

```diff
@@ -1043,6 +1043,219 @@ function Get-JaCoCoReportXml {
+            'branches-valid'   = 0
+            'branches-covered' = 0
+            'branch-rate'      = 1
+            timestamp          = $startTime
+            complexity         = 0
+            version            = '0.1'
+        })
+
+    $packagesElement = Add-XmlElement -Parent $coverageElement -Name 'packages'
+
+    foreach ($package in $packageList) {
+        $packageRelativePath = Get-RelativePath -Path $package.Name -RelativeTo $commonParent
+
+        # "" in root and "sub-dir" elsewhere
+        $packageName = if ($null -eq $packageRelativePath -or "" -eq $packageRelativePath) {
+            ""
+        }
+        else {
+            $packageRelativePathFormatted = $packageRelativePath.Replace("\", "/")
+            $packageRelativePathFormatted
+        }
+
+        $packageElement = Add-XmlElement -Parent $packagesElement -Name 'package' -Attributes ([ordered]@{
+                name          = ($packageName -replace "/$", "")
+                'line-rate'   = if ($package.Line.Total) { $package.Line.Covered / $package.Line.Total } else { 1 }
+                'branch-rate' = 1
+            })
+        $classesElement = Add-XmlElement -Parent $packageElement -Name 'classes'
+
+        foreach ($file in $package.Classes.Keys) {
+            $class = $package.Classes.$file
+            $classElementRelativePath = (Get-RelativePath -Path $file -RelativeTo $commonParent).Replace("\", "/")
+            $classElementName = "$classElementRelativePath"
+            $classElementName = $classElementName.Substring(0, $($classElementName.LastIndexOf(".")))
+            $classElement = Add-XmlElement -Parent $classesElement -Name 'class' -Attributes ([ordered]@{
+                    name          = $classElementName
+                    filename      = $classElementRelativePath
+                    'line-rate'   = if ($class.Line.Total) { $class.Line.Covered / $class.Line.Total } else { 1 }
+                    'branch-rate' = 1
+                })
+            $methodsElement = Add-XmlElement -Parent $classElement -Name 'methods'
+
+            foreach ($function in $class.Methods.Keys) {
+                $method = $class.Methods.$function
+                $methodElement = Add-XmlElement -Parent $methodsElement -Name 'method' -Attributes ([ordered]@{
+                        name      = $function
+                        hits      = $method.Method.Covered
+                        signature = '()'
+                    })
+
+                $linesElement = Add-XmlElement -Parent $methodElement -Name 'lines'
+
+                foreach ($line in $class.Lines.Keys) {
+                    $null = Add-XmlElement -Parent $linesElement -Name 'line' -Attributes ([ordered]@{
+                            number = $line
+                            hits   = $class.Lines.$line.Instruction.Covered
+                        })
+                }
```

Comment:
Lines should be filtered by function name? This currently duplicates all lines for every method